src/net.c

```diff
@@ -374,6 +374,25 @@ int valkeyTcpSetTimeout(valkeyContext *c, const struct timeval tv) {
     return VALKEY_OK;
 }
 
+/* Until glibc 2.39, getaddrinfo with hints.ai_protocol of IPPROTO_MPTCP leads error.
+ * Use hints.ai_protocol IPPROTO_IP (0) or IPPROTO_TCP (6) to resolve address and overwrite
+ * it when MPTCP is enabled.
+ * Ref: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/tools/testing/selftests/net/mptcp/mptcp_connect.c
+ */
+static int valkeyContextSetMptcp(valkeyContext *c, int ai_protocol) {
+    if (c->flags & VALKEY_MPTCP) {
+#ifdef IPPROTO_MPTCP
+        (void)(c);
+        return IPPROTO_MPTCP;
+    }
+#else
+        valkeySetErrorFromErrno(c, VALKEY_ERR_PROTOCOL, "MPTCP is not supported on this platform");
+        return VALKEY_ERR;
+    }
+#endif
+    return ai_protocol;
+}
```

Comment:
Same discussion as we had in the Valkey PR https://github.com/valkey-io/valkey/pull/1811

Let's change this in a similar way?

```suggestion
static int valkeyContextTcpGetProtocol(int is_mptcp_enabled) {
#ifdef IPPROTO_MPTCP
    return is_mptcp_enabled ? IPPROTO_MPTCP : IPPROTO_TCP;
#else
    return IPPROTO_TCP;
#endif
}
```